apps/fishing-map-e2e/src/e2e/map/eez-download.cy.ts

```diff
@@ -0,0 +1,217 @@
+import {
+  GROUPBY_DAY,
+  GROUPBY_FLAG,
+  GROUPBY_FLAG_GEAR,
+  GROUPBY_GEAR,
+  GROUPBY_MMSI,
+  GROUPBY_MONTH,
+  MAP_POPUP_EEZ_SECTION,
+  REPORT_FORMAT_CSV,
+  REPORT_FORMAT_JSON,
+  SIDEBAR_TOGGLE_EEZ,
+} from '../../constants/buttons'
+import { REPORT_CSV_COLUMNS, SEARCH_EEZ, SEARCH_EEZ_FULL_NAME } from '../../constants/search'
+import { API_URL_4WINGS_REPORT, API_URL_GALAPAGOS_INFO, URL_ONE_MONTH } from '../../constants/urls'
+import {
+  deleteDownloadsFolder,
+  disablePopups,
+  getDownloadsFolderPath,
+  getMapCanvas,
+  getQueryParam,
+  scrollSidebar,
+  waitForMapLoadTiles,
+  waitForSidebarLoaded,
+} from '../../support/app.po'
+
+// Static names and paths used for verifications
+const start = getQueryParam(URL_ONE_MONTH.substring(1), 'start').replaceAll(':', '_')
+const end = getQueryParam(URL_ONE_MONTH.substring(1), 'end').replaceAll(':', '_')
+const filename = `${SEARCH_EEZ_FULL_NAME} - ${start},${end}`
+const zipFilename = `${filename}.zip`
+const jsonFilename = `${filename}.json`
+const zipPath = `${getDownloadsFolderPath()}/${zipFilename}`
+const jsonPath = `${getDownloadsFolderPath()}/${jsonFilename}`
+const DATASET = 'public-global-fishing-effort:v20201001'
+const folderToUse = '/layer-activity-data-0/public-global-fishing-effort-v20201001'
+
+const selectOptions = (format: string, attribute: string, time: string) => {
+  cy.getBySel(format).click()
+  cy.getBySel(attribute).click()
+  cy.getBySel(time).click()
+}
+
+const verifyJson = (attributes: string[]) => {
+  cy.readFile(`${getDownloadsFolderPath()}/${jsonFilename}`).then((json) => {
+    cy.wrap(json).its('entries').should('be.an', 'array')
+    cy.wrap(json)
+      .its('entries')
+      .each((entry) => {
+        cy.wrap(entry).should('have.any.keys', [DATASET])
+        cy.wrap(entry[DATASET]).each((field) => cy.wrap(field).should('have.keys', attributes))
+      })
+  })
+}
+
+const verifyCSV = (attributes: string[]) => {
+  cy.readFile(getDownloadsFolderPath() + '/unzip/' + filename + folderToUse + '.csv').then(
+    (csvFile) => {
+      attributes.forEach((col) => cy.wrap(csvFile).should('contain', col))
+    }
+  )
+}
+
+const verifyFileDownload = (file: string, button: string) => {
+  cy.intercept(API_URL_4WINGS_REPORT).as('downloadReport')
+  cy.getBySel(`download-${button}-button`).should('not.be.disabled').click()
+  // This request takes for me 17s, so I use 30 just in case
+  cy.wait('@downloadReport', { requestTimeout: 40000 })
+
+  //Create the same file name that the component CSVLink is generating
+  cy.readFile(file, { timeout: 15000 })
+}
```

Comment:
Common functions used many times inside a page test could go in a PageObject file (eg: `apps/fishing-map-e2e/src/support/map/eez-download.po.ts`) and import them here.

Some repetitive actions inside tests (like [these](https://github.com/GlobalFishingWatch/frontend/pull/2154/files#diff-adc1e64389f81e8dcfdb8819c4468f695ab2ce022cfbb0c03fce50699dd6942bR147-R157))  could also be refactored and extracted to functions inside the `eez-download.po.ts` and invoked here.